analytic_engine/src/row_iter/merge.rs

```diff
@@ -683,13 +683,13 @@ impl MergeIterator {
         }
 
         let pull_start = Instant::now();
-        let buffered_streams = try_join_all(init_buffered_streams).await?;
         self.metrics.scan_duration += pull_start.elapsed();
-        self.metrics.scan_count += buffered_streams.len();
+        self.metrics.scan_count += init_buffered_streams.len();
 
         // Push streams to heap.
         let current_schema = &self.schema;
-        for buffered_stream in buffered_streams {
+        for buffered_stream in init_buffered_streams {
+            let buffered_stream = buffered_stream.await?;
```

Comment:
The futures in the `init_buffered_streams` is not spawned in another runtime, that is to say, the execution of all the futures will be sequential if use forloop to await. Let's use `FuturesUnordered` here.